File: tests/scalar.lua
return {
    pos = {
        a = 0,
        b = 1<<8 - 1,
        c = 1<<8,
        d = 1<<16 - 1,
        e = 1<<16,
        f = 1<<32 - 1,
        g = 1<<32,
        h = 1<<32 + 1,
    },
    neg = {
        a = -(1<<7),
        b = -(1<<15),
        c = -(1<<31),
        d = -(1<<63),
    },
    boolean = {
        t = true,
        f = false,
    },
    strings = {
        s1 = "",
        s2 = "hello",
        s3 = "hello world!",
    },

    floats = {
        f1 = 0.0,
        f2 = 1.0,
        f3 = 3.1,
        f4 = math.pi,
    },

    a = math.maxinteger,
    b = math.mininteger,
    b1 = math.mininteger+1,
    b2 = math.mininteger+2,
    b3 = math.mininteger+3,
    b4 = math.mininteger+4,
}

File: tests/scalar.c
#include "scalar.h"
const struct t_tests_scalar TESTS_SCALAR = {
    .a = 9223372036854775807LLU,
    .b = -9223372036854775807LL - 1LL,
    .b1 = -9223372036854775807LL,
    .b2 = -9223372036854775806LL,
    .b3 = -9223372036854775805LL,
    .b4 = -9223372036854775804LL,
    .boolean = {.f = false, .t = true},
    .floats = {.f1 = 0.0, .f2 = 1.0, .f3 = 3.1, .f4 = 3.1415926535898},
    .neg = {.a = -128, .b = -32768, .c = -2147483648, .d = -9223372036854775807LL - 1LL},
    .pos =
        {.a = 0, .b = 128, .c = 256, .d = 32768, .e = 65536, .f = 2147483648, .g = 4294967296LLU, .h = 8589934592LLU},
    .strings = {.s1 = "", .s2 = "hello", .s3 = "hello world!"},
};
